internal/cloudapi/v2/compose.go

```diff
@@ -1144,7 +1147,12 @@ func (request *ComposeRequest) GetImageRequests(distroFactory *distrofactory.Fac
 	}
 	var irs []imageRequest
 	for _, ir := range *request.ImageRequests {
-		arch, err := distribution.GetArch(ir.Architecture)
+		reqArch := ir.Architecture
+		// If there is an architecture in the blueprint it overrides the request's arch
+		if len(bp.Arch) > 0 {
+			reqArch = bp.Arch
+		}
```

Comment:
Just overriding the value does not seem like the best solution. We should ensure that the values in the image request and BP match if both are set or produce an error if they don't (as it is an invalid request).

Reply:
No, the value in the blueprint should override the other (or maybe the request should override the optional arch in the bp, but they are not expected to match). This is how it works in the weldr API, if arch is set in a blueprint it overrides the default host arch.

Reply:
The client should ensure a valid API call. This would mean extracting the architecture from the BP and making the correct compose request. The question is whether the architecture property should be added to the BP structure. Specifically, the API is already structured so that the requests are for specific environments and architecture, while BP is agnostic of these properties.

The Cloud API has no notion of the host arch, and everything must be specified in the request.